### ISSUE TITLE
Arbitrary DerivPath instance doesn't account for mixed paths

### ISSUE THREAD

**Issue by tphyahoo** [jprupp/haskoin-core#203]
```
instance Arbitrary ArbitraryHardPath where
    arbitrary =
        ArbitraryHardPath <$> (go =<< listOf genIndex)
      where
        go []     = return Deriv
        go (i:is) = (:| i) <$> go is
```

As a side note, I think the above is easier

```
foldl' (:|) Deriv
```

but of course, we still don't have mixed path with that either.

In my #200 rethink, I made this type of testing code easier by tagging path indexes as hard or soft, something like

```
data Bip32PathIndex = Bip32HardIndex KeyIndex | Bip32SoftIndex KeyIndex
```

You would then have Arbitrary instance for Bip32PathIndex, generate a random list of these, and get the hard path from that.

I think I then started using hard/soft index tags in more places in the code, and it was felt this made things unwieldy compared with current approach.  I'm considering to try again but striking a more delicate balance.

```
instance Arbitrary ArbitraryDerivPath
```

is also affected.

**Comment by tphyahoo** [comment]
https://github.com/tphyahoo/haskoin/tree/hard-follow-soft-tphyahoo

fixes 

```
Arbitrary ArbitraryDerivPath
```

Arbitrary HardPath is still pending

**Comment by tphyahoo** [comment]
on second thought, HardPath is derived as using only hard derivations. So I guess this is correct now, only DerivPath needed arbitrary mixing.

**Comment by tphyahoo** [comment]
Fixed by #204 / https://github.com/tphyahoo/haskoin/commit/a727ff89b1e3f72230cd025d560b25fb5f58df3f
